### ISSUE TITLE
Incorrect multipart form validation

### ISSUE THREAD

**Issue by dimuska139** [gin-gonic/gin#1892]
- go version: go1.12.5 linux/amd64
- gin version (or commit ref): v1.4.0
- operating system: Linux Mint 19 Cinnamon 4.15.0-48-generic

## Description
If i try to bind multipart data to struct (one text field and file field) without file, error on the field with file has type *errors.errorString. So this error type has not information to associate error text with form field. But if text field is empty (not file field), error type is gopkg.in/go-playground/validator.v8/ValidationErrors

```go
type ImageUploadForm struct {
	Email string                `form:"email" binding:"required"`
	Img   *multipart.FileHeader `form:"img" binding:"required"`
}
///
var imgForm ImageUploadForm
if err := c.MustBindWith(&imgForm, binding.FormMultipart); err != nil {
	c.Error(err)
	return
}
```
## Screenshots
![Снимок экрана от 2019-05-10 10-40-40](https://user-images.githubusercontent.com/10630525/57510806-538e0c80-7310-11e9-9541-6efe39320697.png)

![Снимок экрана от 2019-05-10 10-44-51](https://user-images.githubusercontent.com/10630525/57510947-ae276880-7310-11e9-8bdd-1caa187e98bc.png)

**Comment by thinkerou** [comment]
because `Img` params set `required`.

**Comment by dimuska139** [comment]
@thinkerou yes, that's the point of my example. Why if Gin binds multipart data to struct without file it returns *errors.errorString but not gopkg.in/go-playground/validator.v8/ValidationErrors? How to understand in which field the error? But in similar situation with text fields (field has "required" param but submitted form field with name "email" doesn't exist) MustBindWith returns exactly gopkg.in/go-playground/validator.v8/ValidationErrors?
